ethcore/builtin/src/lib.rs

```diff
@@ -83,16 +83,15 @@ impl Pricer for Linear {
 /// alt_bn128 constant operations (add and mul) pricing model.
 struct AltBn128ConstOperations {
 	price: usize,
-	eip1108_transition_at: u64,
-	eip1108_transition_price: usize,
+	eip1108_transition_at: Option<u64>,
+	eip1108_transition_price: Option<usize>,
 }
 
 impl Pricer for AltBn128ConstOperations {
 	fn cost(&self, _input: &[u8], at: u64) -> U256 {
-		if at >= self.eip1108_transition_at {
-			self.eip1108_transition_price.into()
-		} else {
-			self.price.into()
+		match (self.eip1108_transition_at, self.eip1108_transition_price) {
```

Comment:
this became a little more complex unfortunately

However, it will only enable the `eip` iff both fields are specified in the chain spec. Do we want a warning/trace if only one field is specified (probably a bug in the case)?

Reply:
Also this matches are `not DRY` but it becomes messy to add a struct `Eip1108` then we would need another trait object in there.

Reply:
Also this two matches are `not DRY` but it would require a trait object or enum such as `Eip1108` to replace `eip_transition_at` and `eip_price` (price types will be different). I prefer not to do because most of this will be fixed by https://github.com/paritytech/parity-ethereum/pull/11039